pytorch_lightning/trainer/training_loop.py

```diff
@@ -556,34 +548,29 @@ def increment_accumulated_grad_global_step(self):
             self.global_step += 1
         self.total_batch_idx += 1
 
-    def save_train_loop_metrics_to_loggers(self, batch_idx, early_stop_epoch, batch_output):
+    def save_train_loop_metrics_to_loggers(self, batch_idx, batch_output):
         # when metrics should be logged
-        should_log_metrics = batch_idx % self.row_log_interval == 0 or early_stop_epoch
+        should_log_metrics = batch_idx % self.row_log_interval == 0 or self.should_stop
         if should_log_metrics or self.fast_dev_run:
             # logs user requested information to logger
             self.log_metrics(batch_output.batch_log_metrics, batch_output.grad_norm_dic)
 
-    def save_loggers_in_training_loop(self, batch_idx, early_stop_epoch):
+    def save_loggers_in_training_loop(self, batch_idx):
         # when loggers should save to disk
-        should_save_log = (batch_idx + 1) % self.log_save_interval == 0 or early_stop_epoch
+        should_save_log = (batch_idx + 1) % self.log_save_interval == 0 or self.should_stop
         if should_save_log or self.fast_dev_run:
             if self.is_global_zero and self.logger is not None:
                 self.logger.save()
 
-    def check_validation_in_train_loop(self, batch_idx, early_stop_epoch, is_last_batch):
+    def check_validation_in_train_loop(self, batch_idx, is_last_batch):
         # decide if we should run validation
         is_val_check_batch = (batch_idx + 1) % self.val_check_batch == 0
         can_check_epoch = (self.current_epoch + 1) % self.check_val_every_n_epoch == 0
         can_check_val = not self.disable_validation and can_check_epoch
-        should_check_val = is_val_check_batch or early_stop_epoch
+        should_check_val = is_val_check_batch or self.should_stop
         should_check_val = should_check_val or (is_last_batch and self.val_check_batch == float('inf'))
         should_check_val = can_check_val and should_check_val
 
-        # if we need to run validation, then also call the checkpoint callback
-        if self.fast_dev_run or should_check_val:
-            self.run_evaluation(test_mode=self.testing)
-            self.call_checkpoint_callback()
-
```

Comment:
@jeremyjordan The tests fail because the evaluation loop is not getting called after the epoch. Did you intend to movie it somewhere else?

Reply:
ahh i think i only meant to remove the `call_checkpoint_callback()`, this was my mistake - glad you caught that!